exchanges/kraken/kraken.go

```diff
@@ -119,15 +134,26 @@ func (k *Kraken) GetAssets() (map[string]*Asset, error) {
 }
 
 // GetAssetPairs returns a full asset pair list
-func (k *Kraken) GetAssetPairs() (map[string]*AssetPairs, error) {
+// Parameter 'info' only supports 3 strings: "fees", "leverage", "margin"
+func (k *Kraken) GetAssetPairs(assetPairs []string, info string) (map[string]AssetPairs, error) {
 	path := fmt.Sprintf("%s/%s/public/%s", k.API.Endpoints.URL, krakenAPIVersion, krakenAssetPairs)
-
+	params := url.Values{}
+	var assets string
+	if len(assetPairs) != 0 {
+		assets = strings.Join(assetPairs, ",")
+		params.Set("pair", assets)
+	}
 	var response struct {
-		Error  []string               `json:"error"`
-		Result map[string]*AssetPairs `json:"result"`
+		Error  []string              `json:"error"`
+		Result map[string]AssetPairs `json:"result"`
 	}
-
-	if err := k.SendHTTPRequest(path, &response); err != nil {
+	if info != "" {
+		if (info != "margin") && (info != "leverage") && (info != "fees") {
+			return response.Result, fmt.Errorf("parameter info can only be 'asset', 'margin' or 'fees'")
+		}
```

Comment:
Given that you can actually pass in 4 types of strings (`info`, `margin`, `leverage`, `fees`), I think its best to remove this code as if the API every expands to more types like it has here, then we need to update this API implementation every time they add a new one. Let Kraken return the error, rather than us